src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
         raise CycloneDXParserError(f"Unable to locate any BOM files at: {str(file_path.parent)}.")
 
+    # We use the `cyclonedx-python-library` library for deserialization following the example here:
+    # https://cyclonedx-python-library.readthedocs.io/en/v7.3.4/examples.html
     with open(file_path, encoding="utf8") as file:
+        json_data = file.read()
+        if defaults.getboolean("dependency.resolver", "validate", fallback=True):
+            try:
+                schema_version = defaults.get("dependency.resolver", "schema", fallback="1.6")
+                my_json_validator = JsonStrictValidator(SchemaVersion.from_version(schema_version))
+            except ValueError as error:
+                raise CycloneDXParserError(f"Unable to find schema validator for {schema_version}: {error}") from error
+            try:
+                validation_errors = my_json_validator.validate_str(json_data)
+            except MissingOptionalDependencyException as error:
+                # Allow MissingOptionalDependencyException.
+                logger.debug("JSON-validation was skipped due to %s", error)
+            except json.JSONDecodeError as error:
+                raise CycloneDXParserError(f"BOM file is invalid: {error}") from error
+
+            if validation_errors:
+                logger.debug("BOM file is invalid: %s", repr(validation_errors))
+                raise CycloneDXParserError(f"BOM file is invalid: {repr(validation_errors)}")
+
+            logger.debug("Successfully validated the BOM file at %s", file_path)
+
         try:
-            return dict(json.load(file))
-        except (ValueError, json.JSONDecodeError) as error:
+            # Mypy complains that `"type[Bom]" has no attribute "from_json"`.
+            # Based on the example provided by cyclonedx-python-lib this type issue needs to be suppressed.
+            # This method is injected into the Bom class that is annotated by ``serializable`` but mypy is not
+            # able to detect that.
+            bom_from_json = Bom.from_json(json.loads(json_data))  # type: ignore[attr-defined]
+        except (ValueError, AttributeError, json.JSONDecodeError) as error:
             raise CycloneDXParserError(f"Could not process the dependencies at {file_path}: {error}") from None
 
+        if isinstance(bom_from_json, Bom):
+            logger.debug("Successfully deserialized the BOM file: %s", repr(bom_from_json))
+            return bom_from_json
+
+        raise CycloneDXParserError(f"Could not process the dependencies at {file_path}")
+
+
+class DependencyTools(str, Enum):
+    """Dependency resolvers supported by Macaron."""
+
+    CYCLONEDX_MAVEN = "cyclonedx-maven"
+    CYCLONEDX_GRADLE = "cyclonedx-gradle"
+    CYCLONEDX_PYTHON = "cyclonedx_py"
+
+
+class DependencyInfo(TypedDict):
+    """The information of a resolved dependency."""
+
+    purl: PackageURL
+    url: str
+    note: str
+    available: SCMStatus
+
+
+class DependencyAnalyzer(ABC):
```

Comment:
Is it okay for me to assume these details:
- `DependencyAnalyzer` is relocated from `dependency_resolver.py`, with **no change** to its behavior
- Old functions like `get_root_component`, `get_deps_components`, `get_deps_from_sbom` and `convert_components_to_artifacts` are moved inside this class with **no change** to their behaviors.